test/src/glossary/widgets/filter_by_test.dart

```diff
@@ -0,0 +1,279 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:kana_to_kanji/src/core/constants/jlpt_level.dart';
+import 'package:kana_to_kanji/src/core/constants/knowledge_level.dart';
+import 'package:kana_to_kanji/src/glossary/widgets/filter_by.dart';
+
+import '../../../helpers.dart';
+
+void main() {
+  group("FilterBy", () {
+    testWidgets("Default view", (WidgetTester tester) async {
```

Comment:
Les tests de ce fichier pourrait être optmi avec une for loop, comme ça:

```dart
const List<Map<String, dynamic>> cases = [
  {"name": "default view", "selectedJlptLevel": [], "selectedKnowledgeLevel": [], "indexesTrue": [0]} // Test case 1
];

for (Map<String, dynamic> testCase in cases) {
  testWidgets(testCase["name"], (WidgetTester tester) async {
    await tester.pumpLocalizedWidget(FilterBy(
      filterGlossary: () => {},
      selectedJlptLevel: testCase["selectedJlptLevel"],
      selectedKnowledgeLevel: testCase["selectedKnowledgeLevel"],
    ));
    await tester.pumpAndSettle();

    final checkboxes = find.byType(Checkbox);
    expect(checkboxes, findsNWidgets(9));

    for(int i=0; i < 9; i++) {
      expect((tester.widget(checkboxes.at(i)) as Checkbox).value, (testCase["indexesTrue"] as List<int>).contains(i) ? isTrue:isFalse);
    }
  });
}
```

Reply:
Fait